addon/components/transition.js

```diff
@@ -0,0 +1,72 @@
+import Component from '@glimmer/component';
+import { tracked } from '@glimmer/tracking';
+import { use } from 'ember-could-get-used-to-this';
+import { action } from '@ember/object';
+import { assert } from '@ember/debug';
+import {
+  AppliedClassNamesResource,
+  TransitionVisibilityResource,
+} from '../resources/transition';
+
+export default class TransitionComponent extends Component {
+  constructor() {
+    super(...arguments);
+
+    let { show } = this.args;
+
+    assert(ARG_SHOW_MISSING_ERROR_MESSAGE, typeof show === 'boolean');
+
+    // Normally this value is computed lazily, but we *do not want this*
+    // If we allow the value to be lazy, the class names are first computed only once the component is rendered
+    // Simply accessing the property will cause the resource to begin computing
+    this.transitionClassNames;
```

Comment:
I want to play around with the implementation some more and see if this can be avoided.

The tricky bit here is that we want the Resource that backs the class name string to start working right away, as soon as the component is rendered. This is so that when we change `@show` from `false` to `true`, we can respond to that and perform the transition as expected.

When we allow this value to be lazy, we lose the ability to respond to the transition; from the perspective of the Resource, it appears that we're performing our initial render with a `@show` value that is `true`, when in reality the initial render took place earlier with a `@show` value of `false`.

The Resource is initialized only once `@show` becomes `true` for the first time because, without this code that is eagerly forcing it to be instantiated, we only access the class string the first time we actually render the component contents for the first time.

An alternative that I want to explore is having the `Transition` component monitor whether we're in our first render or not. This context can be passed along to the Resource.